modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/onboarding/components/SubscriptionPlanRow.kt

```diff
@@ -0,0 +1,327 @@
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.ripple
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.ReadOnlyComposable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import au.com.shiftyjelly.pocketcasts.compose.AppThemeWithBackground
+import au.com.shiftyjelly.pocketcasts.compose.components.TextH30
+import au.com.shiftyjelly.pocketcasts.compose.components.TextH50
+import au.com.shiftyjelly.pocketcasts.compose.components.TextP40
+import au.com.shiftyjelly.pocketcasts.compose.preview.ThemePreviewParameterProvider
+import au.com.shiftyjelly.pocketcasts.compose.theme
+import au.com.shiftyjelly.pocketcasts.payment.BillingCycle
+import au.com.shiftyjelly.pocketcasts.payment.SubscriptionPlan
+import au.com.shiftyjelly.pocketcasts.payment.SubscriptionTier
+import au.com.shiftyjelly.pocketcasts.ui.theme.Theme.ThemeType
+import au.com.shiftyjelly.pocketcasts.images.R as IR
+import au.com.shiftyjelly.pocketcasts.localization.R as LR
+
+@Composable
+fun AvailablePlanRow(
+    plan: SubscriptionPlan.Base,
+    isSelected: Boolean,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    SubscriptionPlanRow(
+        plan = plan,
+        isSelected = isSelected,
+        onClick = onClick,
+        modifier = modifier,
+        badgeConfig = BadgeConfig.availablePlansConfig(),
+    )
+}
+
+@Composable
+fun UpgradePlanRow(
+    plan: SubscriptionPlan.Base,
+    isSelected: Boolean,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    otherPlan: SubscriptionPlan.Base? = null,
+) {
+    SubscriptionPlanRow(
+        plan = plan,
+        isSelected = isSelected,
+        onClick = onClick,
+        modifier = modifier,
+        badgeConfig = BadgeConfig.upgradePlansConfig(calculatedSavingPercent = otherPlan?.let { plan.savingsPercent(otherPlan) }),
+    )
+}
+
+private enum class BadgePosition {
+    RIGHT,
+    CENTER,
+}
+
+private sealed interface BadgeText {
+
+    @Composable
+    fun text(): String
+
+    data class StaticTextResource(
+        @StringRes val stringResId: Int,
+    ) : BadgeText {
+
+        @Composable
+        override fun text() = stringResource(stringResId)
+    }
+
+    data class CalculatedText(
+        val provider: @Composable () -> String,
+    ) : BadgeText {
+        @Composable
+        override fun text() = provider()
+    }
+}
+
+private data class BadgeConfig(
+    val badgeText: BadgeText,
+    val badgePosition: BadgePosition,
```

Comment:
I think we should add three things here:

- `verticalPadding: Dp`: For available plans, it would be `0.dp`, and for upgrade plans, `4.dp`. This is an additional padding to the `Row`.
- `labelSpacing: Dp`: For available plans it would be `0.dp`, and for upgrade plans `4.dp`. This is for the `Column` with labels and can be applied with something like `Arrangement.spacedBy(labelSpacing)`.
- `priceFormatting: Enum`: For available plans we display "price per month"m and for upgrade plans it's "price per week".